UI/src/view/point_asset/massTransitStopForm.js

```diff
@@ -77,15 +81,24 @@
         });
       } else {
         if(optionalSave()){
-          new GenericConfirmPopup('Oletko varma, ettet halua lähettää pysäkin tietoja Tierekisteriin? Jos vastaat kyllä, tiedot tallentuvat ainoastaan OTH-sovellukseen', {
-            successCallback: function () {
-              selectedMassTransitStopModel.setAdditionalProperty('trSave', [{ propertyValue: 'false' }]);
-              saveStop();
-            },
-            closeCallback: function () {
-              saveStop();
-            }
-          });
+          if(saveNewBusStopStrategy()) {
+            new GenericConfirmPopup('Koska tämä bussipysäkki on määritetty vihjeeksi, siihen liittyviä tietoja ei lähetetä Tierekisteriin. Haluatko silti tallentaa sen OTH:ssa?', {
+              successCallback: function () {
+                selectedMassTransitStopModel.setAdditionalProperty('trSave', [{ propertyValue: 'false' }]);
+                saveStop();
+              }});
+          } else {
+            new GenericConfirmPopup('Oletko varma, ettet halua lähettää pysäkin tietoja Tierekisteriin? Jos vastaat kyllä, tiedot tallentuvat ainoastaan OTH-sovellukseen', {
+              successCallback: function () {
+                selectedMassTransitStopModel.setAdditionalProperty('trSave', [{ propertyValue: 'false' }]);
+                saveStop();
+              },
+              closeCallback: function () {
+                saveStop();
+              }
+            });
+          }
+
```

Comment:
I don't think you can. The popups are different not only in the message but its buttons meanings (you can translate the messages to see what I meant). Only successCallback is equal on both. And one of them have a closecallback(because the button meaning).